### ISSUE TITLE
GUI: add standard global PVs to instrument database

### ISSUE THREAD

**Issue by kjwoodsISIS** [kjwoodsISIS/IBEX#708]
Currently you can only browse PVs from IOCs in the GUI when adding a block - PVs for items such as beam current, moderator and shutter status need to be added by hand. This ticket is to add the central beamlogger PVs to the PV database, maybe by updating them at start_inst time from a list  

may need a bit of thought as currently PVs are tied to IOCs in the database schema.

**Comment by kjwoodsISIS** [comment]
_Imported from trac issue 709.  Created by faa59 on 2015-04-09T16:42:16, last modified: 2016-01-26T15:22:28_

**Comment by kjwoodsISIS** [comment]
_Trac comment by skn09965 on 2016-01-07 17:34:05:_

From: ICP [mailto:icp-bounces@svr2.nd.rl.ac.uk] On Behalf Of kathryn.baker@stfc.ac.uk
Sent: 23 December 2015 12:12
To: icp@svr2.nd.rl.ac.uk
Subject: [ICP] Story 709 - standard global PVs

Hi all,

Emailing this as I will forget by January otherwise.

I’ve got a solution for this, which is maybe not the tidiest, but works most readily with our existing system.

I have created an extra table with the standard PVs – otherwise the database server/blockserver gets unhappy as it can’t check on the IOC. As an extra table it is fairly straightforward to just create and populate it (run the updated database server without the table it does complain, because it is trying to read it). I’ll put the script in with the other database creation bits. I’ve also added all the shutter information at the moment – do we want that in there? It should also be possible to maintain the contents centrally in time either via remote access to the database, or just drop it and repopulate it on every start_inst (which seems overkill), or write a small program in something to check for a record and add it if it doesn’t exist based on the contents of a file.

I’ve updated the database server to read this new table and provide them as a standard JSON string separate to the main PV list, which just tags them on the end (without doing that they disappear from the GUI).

I’ve updated the GUI to include a lookup for the PVS:STANDARD pv that is now added to the database server, and added “Standard PVs” to the options of PVs to view in the PV selection criteria. There is one change I haven’t tried to implement yet, and I’m not sure if it should be dealt with as a new ticket – the person selecting from the standard pvs has to remember to uncheck the ‘local’ box for the block, otherwise it start prepending the instrument prefix. It should be possible to stop that default, or overwrite it, I think.

I don’t know if anyone else can see a better architecture, but I am quite happy to alter this.

The one thing I’m not sure about is the naming. At the moment I’ve put it all through as variations on “standard pvs” but I’m not sure that is the most usefully descriptive term. However, I’m not sure what else would make sense. All PVs are global, so don’t really want to call them that, so is standard descriptive enough? Basically these are PVs not associated with an IOC running on whichever instrument you are, but that you might want to access. If there are no better ideas I’ll leave it as standard.

Kathryn
